In [2]:
# data read Imports
import pandas as pd

# Machine Learning Imports
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split

# For evaluating our ML results
from sklearn import metrics

In [136]:
df = pd.read_csv("data/sample.csv")
df

female  read  write  math  pass  femalexmath
0         0    57     52    41     0            0
1         1    68     59    53     0           53
2         0    44     33    54     0            0
3         0    63     44    47     0            0
4         0    47     52    57     0            0
..      ...   ...    ...   ...   ...          ...
195       1    55     59    52     0           52
196       1    42     46    38     0           38
197       1    57     41    57     0           57
198       1    55     62    58     1           58
199       1    63     65    65     1           65

[200 rows x 6 columns]

In [123]:
feature = df[['female', 'math']]
target = df['pass']

In [128]:
# scikit learn
lr = LogisticRegression()
lr.fit(feature,target)
print("coef : ",lr.coef_ , " \nintercept: ", lr.intercept_)

coef :  [[0.8245473  0.16223285]]  
intercept:  [-10.60588574]


In [129]:
target.value_counts()

0    151
1     49
Name: pass, dtype: int64

In [149]:
import math
print("female : ", math.exp(lr.coef_[0][0])," math :",math.exp(lr.coef_[0][1])," intercept:" ,math.exp(lr.intercept_[0]))

female :  2.2808479910448938  math : 1.1761340682855113  intercept: 2.4769791355793902e-05


## Interpretation 

- If the gender is female then the odds of passing exam is 2.28. When other value fixed.
- If the math number increase by one unit then the odds of passing exam is 1.176. When other value fixed.
- When all numerical features are zero and the categorical features are at the reference category, the estimated odds are 2.47e-05 . The interpretation of the intercept weight is usually not relevant.

In [137]:
df.groupby("female")['pass'].value_counts()

female  pass
0       0       74
        1       17
1       0       77
        1       32
Name: pass, dtype: int64

> # Task #1
- Do Logistic Regression for `female, read, write, math` columns 
- Print coef and intercept 
- Write Interpretation 
- And send me the notebook not screenshot at nurulaktertowhid@gmail.com

